plugin-server/src/main/ingestion-queues/session-recording/services/console-logs-ingester.ts

```diff
@@ -145,9 +145,13 @@ export class ConsoleLogsIngester {
             })
         }
     }
+
     public async start(): Promise<void> {
         const connectionConfig = createRdConnectionConfigFromEnvVars(this.serverConfig)
+
         const producerConfig = createRdProducerConfigFromEnvVars(this.serverConfig)
+        producerConfig['KAFKA_PRODUCER_BATCH_SIZE'] = this.serverConfig.CONSOLE_LOG_INGESTER_PRODUCER_BATCH_SIZE
```

Comment:
I'm a bit surprised with having several producers in the same process, what are the expected benefits from it? I'm worried that it will make it harder to reason about producer logs. It should be fine to increase the batch size via `KAFKA_PRODUCER_BATCH_SIZE` on all topics for this consumer.

I'm not sure that the batch size will help though: the issue is that the producer queue is full. `queue.buffering.max.kbytes` and `queue.buffering.max.messages` are what we should be tuning, [see rdk doc](https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md), but the defaults are already pretty generous: 100k messages, 64MiB. 

If the queue is full, it would be OK for the `produce` call to block for a while (implementing some backpressure), but it looks like node-rdkafka does not allow this. Should we implement this in `KafkaProducerWrapper`?

Reply:
> I'm a bit surprised with having several producers in the same process, what are the expected benefits from it?

It wasn't a conscious choice... it's normal to share a producer?

Reply:
Yes, analytic consumers just share the `hub.kafkaProducer` one. We probably will need one per destination cluster though.

Reply:
I'll try bumping the number of messages in `queue.buffering.max.messages`

And separately switch to one producer... 

I bet we can de-duplicate the produced messages too since we're only using them for search emitting 1 is just as valid as 100 if the client is logging in a loop and it does look like we get relatively high peaks

Reply:
Worth doing some queries on the log_entries table to try and find pathological cases indeed.

Otherwise, we'll want to rework the loop to add some checks on the number of pending promises, await them when a threshold is reached before processing more messages. I have not implemented this on analytics as there's not that much message amplification between input and output

Reply:
Also, bigger write batches could indeed help with throughput, if the MSK brokers support it. But they won't stop a runaway producer if the issue is too many logs per rrweb snapshot.

Reply:
Switched to the suggested queue param so I can test if setting that higher for blobby helps in the short term